lib/portlayer/storage/storage.go

```diff
@@ -12,6 +12,74 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// Package storage provides abstractions for interacting with storage locations, both management and access.
+// There are two primary type of storage handled by this package at this time:
+// a. VMDKs on a vSphere datastore
+// b. NFS shares
+//
+// We have more than two sub-packages however as we use distinct packages to provide different lifecycle and
+// usage semantics for types of data using those mechanisms. Examples include:
+// * application logic (images)
+// * application transient data (image read/write layer)
+// * application persistent data (volumes)
+//
+// As such there are a set of concepts captured as interfaces:
```

Comment:
nit: it's a little confusing to say that this is a list of interfaces, include store, and then explain in the following paragraph that store isn't an interface